bin/console

```diff
@@ -1,8 +1,13 @@
 #!/usr/bin/env php
 <?php
 
+declare(strict_types=1);
```

Comment:
tohle tusim zapisujem na 1. radek a melo by to kontrolovat CI

Reply:
U tohoto je jedno, na jaký řádek se to zapisuje pokud vím. Jediné omezení je, že to musí být to první, co následuje po <?PHP

Reply:
Pusť si kontrolu přes `composer qa`, neprojde to. PHP sice vyžaduje jen na začátku a nic předtím, my jsme ale zavedli coding-standard  kde je to vyžadováno ve tvaru `<?php declare(strict_types = 1);` na prvním řádku

Reply:
composer qa mi ani nejde spustit. Navíc jsem to ani nikdy nepoužíval, takže netuším, o co přesně jde. Počítám, že je to nějaký checker kódu. Pokud jste si pravidla psali sami, tak proč to není ověřené například pomocí tohoto regexu: _<\?php\s+declare\(strict_types = 1\);.*_ Ono totiž některé editory to dávají až pod tak jak to mám já, někteří programátoři dostanou už hotovou třídu, tu překopírují a nezajímá je kde to je, někteří neznají Váš codding standard a prostě to dají tak jak všichni ostatní takto o řádek či dva níže. Pak mi přijde, že jsou z toho zbytečné faily ne?

Reply:
Jestliže jsi na Windows, spust to přes WSL (Windows Subsystem for Linux), kde to jde.

Když se podíváš do composer.json, zjistíš že jde o composer scripty, které spouští nástroje jako PHPStan, PHP_CodeSniffer a linter. 

Ano, každý píšeme trochu jinak a máme jinak nastavený editor (proto `.editorconfig`) a když jde o šablony souborů například, tak se většina syntaktických problémů dá vyřešit pomocí `composer csf`.
Ruleset je definovaný v `ruleset.xml`. Myslím si že každý větší projekt už dnes používá jak codesniffer tak nějaký nástroj pro statickou analýzu kódu (PHPStan, Psalm nebo jiný)

Reply:
Dobré vědět. .editorconfig to musím nějak říct netbeans, že to tam je? Zdá se, že to ignoruje.
composer csf neznám naučím se díky za tip.
ruleset.xml napadlo mě to, ale studovat jsem to nechtěl.
Asi se používají codesniffery a statické analýzy. Já se s tím však bohužel nesetkal ještě. :) Nicméně to vypadá jako dobrá fičura, když to funguje jak člověk chce.

Reply:
.editorconfig, v PHPStormu to mám přes rozšíření, myslím si že pro NetBeans by mohlo být taky, ale nejsem si jistý.

`composer csf` je jen v tomto projektu, koukni do `composer.json` do sekce **scripts**. Viz dokumentace composeru https://getcomposer.org/doc/articles/scripts.md

Statická analýza super, najde to spoustu bugů :) A codesniffer vymáhá nějaká codestyle v rámci projektu + umí to odhlait nějaké nedefinované proměnné apod., ale to zvládá většinou  i statická analýza.